gatling-http/src/main/scala/io/gatling/http/check/ws/WsFrameCheck.scala

```diff
@@ -30,7 +32,8 @@ sealed trait WsFrameCheck {
 }
 
 object WsFrameCheck {
-  final case class Binary(name: String, matchConditions: List[WsCheck.Binary], checks: List[WsCheck.Binary], isSilent: Boolean) extends WsFrameCheck {
+  final case class Binary(checkName: Expression[String], matchConditions: List[WsCheck.Binary], checks: List[WsCheck.Binary], isSilent: Boolean, name: String)
```

Comment:
I don't like it. I'd rather have 2 sets of types: one for the DSL with the Expression and one for runtime with the resolved value.

Reply:
> how similar logic was implemented for queryParamMap / queryParamSeq and decided to try it the same way. I found this to be a simpler solution...

Where is that?
`queryParam` uses 2 different sets:
* `io.gatling.http.request.builder.HttpParam` with unresolved Expressions
* `io.gatling.http.client.Param` for resolved Strings

See https://github.com/gatling/gatling/blob/main/gatling-http/src/main/scala/io/gatling/http/request/builder/package.scala#L62